### PR TITLE
I18n localize - compile-time

### PR DESCRIPTION
**Pull request by petebacondarwin**
Blocked on https://github.com/angular/angular/pull/32594

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: integration/cli-hello-world-ivy-compat/src/polyfills.ts [hunk 1]
```diff
@@ -63,9 +63,11 @@
  *
  * The following flags will work for all browsers.
  *
- * (window as any).__Zone_disable_requestAnimationFrame = true; // disable patch requestAnimationFrame
+ * (window as any).__Zone_disable_requestAnimationFrame = true; // disable patch
+ * requestAnimationFrame
  * (window as any).__Zone_disable_on_property = true; // disable patch onProperty such as onclick
- * (window as any).__zone_symbol__BLACK_LISTED_EVENTS = ['scroll', 'mousemove']; // disable patch specified eventNames
+ * (window as any).__zone_symbol__BLACK_LISTED_EVENTS = ['scroll', 'mousemove']; // disable patch
+ * specified eventNames
```
Comment (ocombe) [inline integration/cli-hello-world-ivy-compat/src/polyfills.ts]: you don't need those changes ^

**Comment by ocombe** [comment]
Note: all the commits up to `test: make cli-hello-world-ivy-minimal/debug-test.sh executable` are part of the other unmerged PR: https://github.com/angular/angular/pull/31609 and should not be reviewed here

File: packages/localize/runtime/src/translate.ts [hunk 2]
```diff
@@ -81,81 +49,5 @@ export function clearTranslations() {
...
- * Make an array of `cooked` strings that also holds the `raw` strings in an additional property.
- *
- * @param cooked The actual values of the `messagePart` strings.
- * @param raw The original raw values of the `messagePart` strings, before escape characters are
- * processed.
- */
-function makeTemplateObject(cooked: string[], raw: string[]): TemplateStringsArray {
-  Object.defineProperty(cooked, 'raw', {value: raw});
-  return cooked as any;
+  return _translate($localize.TRANSLATIONS, messageParts, substitutions);
```
Comment (ocombe) [inline packages/localize/runtime/src/translate.ts]: Any reason why you're just calling another function but with the arguments in another order? Shouldn't you remove one of them and refactor the code?

File: packages/localize/src/utils/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * The character used to mark the start and end of a placeholder name.
+ */
+export const PLACEHOLDER_NAME_MARKER = ':';
+
+/**
+ * Translate the text of the given message, using the given translations.
+ *
+ * The translation may reorder (or remove) substitutions as appropriate.
```
Comment (ocombe) [inline packages/localize/src/utils/src/utils.ts]: I feel like this needs more documentation: explain the arguments and give a small example if possible

File: packages/localize/src/utils/src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      ]);
+  return result;
+}
+
+/**
+ * Parse the messageParts and placeholderNames out of a translation key.
+ *
+ * @param translationKey the message to be parsed.
+ */
+export function parseTranslation(translationKey: string): ParsedTranslation {
```
Comment (ocombe) [inline packages/localize/src/utils/src/utils.ts]: should we export those functions (in util.ts) as private apis that can be accessed from the outside?

File: integration/cli-hello-world-ivy-compat/src/polyfills.ts (hunk 1 above)
Comment (petebacondarwin) [reply integration/cli-hello-world-ivy-compat/src/polyfills.ts]: Agh! Damn auto-format.

File: packages/localize/runtime/src/translate.ts (hunk 2 above)
Comment (petebacondarwin) [reply packages/localize/runtime/src/translate.ts]: The inner function also requires us to provide the `translations` object, which is not part of the outer API.

File: packages/localize/src/utils/src/utils.ts (hunk 4 above)
Comment (petebacondarwin) [reply packages/localize/src/utils/src/utils.ts]: Indeed!

File: packages/localize/src/utils/src/utils.ts (hunk 3 above)
Comment (petebacondarwin) [reply packages/localize/src/utils/src/utils.ts]: The underlying function in `packages/localize/src/utils/translations.ts` has more information now.

**Comment by petebacondarwin** [comment]
Closing in favour of https://github.com/angular/angular/pull/32881

**Pull request closed without merging**